ecs-cli/modules/cli/cluster/cluster_app_test.go

```diff
@@ -224,6 +224,114 @@ func TestClusterUpWithAvailabilityZones(t *testing.T) {
 	assert.NoError(t, err, "Unexpected error bringing up cluster")
 }
 
+func TestClusterUpWithCustomRole(t *testing.T) {
+	defer os.Clearenv()
+	mockECS, mockCloudformation := setupTest(t)
+
+	instanceRole := "sparklepony"
+
+	gomock.InOrder(
+		mockECS.EXPECT().Initialize(gomock.Any()),
+		mockECS.EXPECT().CreateCluster(clusterName).Return(clusterName, nil),
+	)
+
+	gomock.InOrder(
+		mockCloudformation.EXPECT().Initialize(gomock.Any()),
+		mockCloudformation.EXPECT().ValidateStackExists(stackName).Return(errors.New("error")),
```

Comment:
Note: these expectations are correct in the happy path, but not in the error cases. They have been duly removed from the tests below.